### PR TITLE
Fix namelist creating for multiple instances

### PR DESCRIPTION
**Pull request by blcc**
Get correct NTASK_OCN in buildcpp, which will be NINST_OCN times when run in multiple instance.
Move pg_blom into instance loop to reset it in every instance.

### CODE REVIEW

File: cime_config/buildcpp [hunk 1]
```diff
@@ -38,6 +38,8 @@ def create_dimmod(case):
     blom_vcoord = case.get_value("BLOM_VCOORD")
     ntasks_ocn = case.get_value("NTASKS_OCN")
     objroot = case.get_value("OBJROOT")
+    if case.get_value("NINST_OCN") > 1:
+        ntasks_ocn = case.get_value("NTASKS_PER_INST_OCN")
```
Comment (TomasTorsvik) [inline cime_config/buildcpp]: I would prefer to define `ntasks_ocn` once, not overwrite
```
if (case.get_value("NINST_OCN") > 1) then
   ntasks_ocn = case.get_value("NTASKS_PER_INST_OCN")
else
   ntasks_ocn = case.get_value("NTASKS_OCN")
end if
```

**Review by TomasTorsvik — APPROVED** [review]
@blcc - I think this looks fine, but suggest not to overwrite `ntasks_ocn`.
Can merge this one you have made the change (or explain why you want to keep the code as it is).

File: cime_config/buildcpp (hunk 1 above)
Comment (blcc) [reply cime_config/buildcpp]: You're right, it is more clear with if-else.
I'll send updated PR tomorrow.

**Comment by blcc** [comment]
Hi Tomas,
I just found NTASKS_PER_INST_OCN output same result with NTASKS_OCN when only one instance.
So that I replace it.

**Comment by TomasTorsvik** [comment]
@blcc - I'm not convinced this is a good solution, since users may expect that setting the "NTASKS_OCN" option will set the `ntasks_ocn` variable. We should not change functionality, especially in the CMIP6 version of the model. I think its OK to make `ntasks_ocn` dependant on "NINST_OCN", since I don't think anyone will change "NINST_OCN" without knowing what they are doing.

**Comment by blcc** [comment]
It also make point.
I change it back.

**Comment by TomasTorsvik** [comment]
@blcc - I merged this PR, and have created a new tag `v1.4.3` on the `release-1.4` branch, for reference when you update the  `Externals.cfg` file.

**Comment by blcc** [comment]
Thank you very much.

**Pull request merged**